backend/business-partner-agent/src/main/java/org/hyperledger/bpa/config/RuntimeConfig.java

```diff
@@ -93,8 +96,40 @@ public void onApplicationEvent(StartupTasks.AcaPyReady event) {
                     .statusConfig()
                     .flatMap(c -> c.getAs("tails_server_base_url", String.class))
                     .isPresent();
+
```

Comment:
As the AcaPyConfig is a bean as well it makes more sense to have it implement `ApplicationEventListener<StartupTasks.AcaPyReady>`as well and move all this initialization parts into it. 

Also each `ac.statusConfig()` call is a web request. Its more efficient to get the config once and then map it accordingly.

Reply:
thanks! will clean that up, didn't realize the statusConfig was a new call each time.